pkgs/top-level/all-packages.nix

```diff
@@ -15546,12 +15546,14 @@ with pkgs;
     crystal_1_7
     crystal_1_8
     crystal_1_9
+    crystal_1_11
+    crystal_1_12
     crystal;
 
   crystal2nix = callPackage ../development/compilers/crystal2nix { };
 
   crystalline = callPackage ../development/tools/language-servers/crystalline {
-    llvmPackages = llvmPackages_15;
+    llvmPackages = llvmPackages_18;
```

Comment:
Instead of manually defining it here, why not expose the used llvm through `passthru` on `crystal` and use that? This way they will always be in sync.

Reply:
Good idea! I pushed up a commit to do that just now